giraffe/src/components/Gauge.tsx

```diff
@@ -171,8 +171,17 @@ const drawSegmentedGauge = (
   colors: Color[],
   gaugeSize: number
 ): void => {
+  const thresholdColors = []
+  if (Array.isArray(colors)) {
+    colors.forEach(color => {
+      if (color.value >= minValue && color.value <= maxValue) {
+        thresholdColors.push(color)
+      }
+    })
+  }
+
   const sortedColors = Array.isArray(colors)
-    ? colors.sort((color1, color2) => color1.value - color2.value)
+    ? thresholdColors.sort((color1, color2) => color1.value - color2.value)
```

Comment:
Since `thresholdColors` is a new variable and guaranteed to be an array, we don't need to check if `colors` is an array anymore. We can just say `sortedColors` is the result of thresholdColors.sort().

Reply:
fixed!